include/picolibrary/indicator.h

```diff
@@ -120,6 +121,58 @@ class Fixed_Intensity_Indicator_Concept {
     auto toggle() noexcept -> Result<Void, Error_Code>;
 };
 
+/**
+ * \brief GPIO output pin based fixed intensity indicator
+ *
+ * \tparam Output_Pin The type of output pin being adapted.
+ */
+template<typename Output_Pin>
+class GPIO_Output_Pin_Fixed_Intensity_Indicator : public Output_Pin {
+  public:
+    using Output_Pin::Output_Pin;
+
+    /**
+     * \brief Initialize the indicator's hardware.
+     *
+     * \param[in] initial_indicator_state The initial state of the indicator.
+     *
+     * \return Nothing if indicator hardware initialization succeeded.
+     * \return The error reported by the underlying pin if pin hardware initialization
+     *         failed.
+     */
+    auto initialize( Initial_Indicator_State initial_indicator_state = Initial_Indicator_State::EXTINGUISHED ) noexcept
+    {
+        return Output_Pin::initialize(
+            initial_indicator_state == Initial_Indicator_State::ILLUMINATED
+                ? GPIO::Initial_Pin_State::HIGH
+                : GPIO::Initial_Pin_State::LOW );
+    }
+
+    /**
+     * \brief Illuminate the indicator.
+     *
+     * \return Nothing if illuminating the indicator succeeded.
+     * \return The error reported by the underlying pin if transitioning the pin to the
+     *         high state failed.
+     */
+    auto illuminate() noexcept
+    {
+        return this->transition_to_high();
+    }
+
+    /**
+     * \brief Extinguish the indicator.
+     *
+     * \return Nothing if extinguishing the indicator succeeded.
+     * \return The error reported by the underlying pin if transitioning the pin to the
+     *         low state failed.
```

Comment:
Please use the context of the indicator operation when discussing what failed.

```suggestion
     * \return The error reported by the underlying pin if extinguishing the indicator
     *         failed.
```